src/buffer.c

```diff
@@ -112,30 +113,29 @@ GglError ggl_str_to_int64(GglBuffer str, int64_t *value) {
         return GGL_ERR_INVALID;
     }
 
-    for (; i < str.len; i++) {
-        uint8_t c = str.data[i];
-
-        if ((c < '0') || (c > '9')) {
-            GGL_LOGE("Invalid character %c when parsing int64.", c);
-            return GGL_ERR_INVALID;
-        }
-
-        if (mult_overflow_int64(ret, 10U)) {
-            GGL_LOGE("Overflow when parsing int64 from buffer.");
-            return GGL_ERR_RANGE;
+    for (; i < str.len; i++)
+        CBMC_CONTRACT(assigns(ret, i), decreases(str.len - i)) {
+            uint8_t c = str.data[i];
+
+            if ((c < '0') || (c > '9')) {
+                GGL_LOGE("Invalid character %c when parsing int64.", c);
+                return GGL_ERR_INVALID;
+            }
+
+            if (mult_overflow_int64(ret, 10U)) {
+                GGL_LOGE("Overflow when parsing int64 from buffer.");
+                return GGL_ERR_RANGE;
+            }
+            ret *= 10;
+
+            if (add_overflow_int64(ret, sign * (c - '0'))) {
+                GGL_LOGE("Overflow when parsing int64 from buffer.");
+                return GGL_ERR_RANGE;
+            }
+            ret += sign * (c - '0');
         }
-        ret *= 10;
 
-        if (add_overflow_int64(ret, sign * (c - '0'))) {
-            GGL_LOGE("Overflow when parsing int64 from buffer.");
-            return GGL_ERR_RANGE;
-        }
-        ret += sign * (c - '0');
-    }
-
-    if (value != NULL) {
-        *value = ret;
-    }
```

Comment:
Good catch. Why would this ever be NULL. This should have been an assert :)

Reply:
Earlier, I was having all pointer parameters assume they could be null, since we didn't have checks for it. But now that we have warnings for passing null to args declared non-null, I've been removing some of the NULL handling where it doesn't make sense.